### ISSUE TITLE
Product Availability Batch update

### ISSUE THREAD

**Issue by Engazan** [mallgroup/mpapi-client-php#22]
Zdravím je neaka možnosť ako cez tento Mall wrapper dokážem urobiť "batch update availability"? 

https://marketplaceapiv2.docs.apiary.io/#reference/availability/batch-update-of-productvariant-availability/batch-availability-change-for-products-and-variants

Skúšal som toto, ale jedna chyba za druhou ako "Missing key category_id" a iné ...
```php
$products = new Products($client)

$product = new Product();
$product->setId($sku);
                ->setStatus($status);
                ->setInStock($mainStock);

$variant = new Variant();
$variant->setId($variantId)
               ->setId($variantSku)
               ->setStatus($variantStatus)
               ->setInStock($variantInStock);
$product->addVariant($variant);

$products->add($product)
                  ->put();
```

**Comment by michalsalon-mall** [comment]
Hello,

current v3 of the client does not support batch endpoints. Sorry.

New version 4, with support for batch endpoints, stricter types etc. is currently in development and should be released relatively soon™.

**Comment by MiChAeLoKGB** [comment]
Hi, a new **beta** version with support for Batch update was released: [v4.0.0-beta](https://github.com/mallgroup/mpapi-client-php/tree/v4.0.0-beta)

It's still a work in progress, but should not change much, and any comments and suggestions are appreciated.

**Comment by michalsalon-mall** [comment]
Hi, a new **beta** version with support for Batch update was released: [v4.0.0-beta](https://github.com/mallgroup/mpapi-client-php/tree/v4.0.0-beta)

It's still a work in progress, but should not change much, and any comments and suggestions are appreciated.

**Comment by qwerin** [comment]
> Hi, a new **beta** version with support for Batch update was released: [v4.0.0-beta](https://github.com/mallgroup/mpapi-client-php/tree/v4.0.0-beta)
> 
> It's still a work in progress, but should not change much, and any comments and suggestions are appreciated.

Zatím verze 4 je nepoužitelná chybí tam jedna zásadní funkce a to je vytvořit produkt jinak než z api ->
[private function __construct( ... ](https://github.com/mallgroup/mpapi-client-php/blob/4a7bbe220b4ee9f8f1dfb37952694f8fd6f0cbbe/src/Article/Entity/Product.php#L63)
a ješte k tomu je to [final class](https://github.com/mallgroup/mpapi-client-php/blob/4a7bbe220b4ee9f8f1dfb37952694f8fd6f0cbbe/src/Article/Entity/Product.php#L17) takže bez moznosti rozšíření..
Takže zbývá jen pomocí metody [createFromApi](https://github.com/mallgroup/mpapi-client-php/blob/4a7bbe220b4ee9f8f1dfb37952694f8fd6f0cbbe/src/Article/Entity/Product.php#L135) a to je nic moc pro nějakou slušnou implementaci.

**Comment by michalsalon-mall** [comment]
Hi,

to create a product or variant, the Product/Variant request DTO's should be used:
[ProductRequest](https://github.com/mallgroup/mpapi-client-php/blob/v4.0.0-beta/src/Article/DTO/ProductRequest.php)
[VariantRequest](https://github.com/mallgroup/mpapi-client-php/blob/v4.0.0-beta/src/Article/DTO/VariantRequest.php)

If you try to call [createProduct](https://github.com/mallgroup/mpapi-client-php/blob/328c4fcc22544f9cede2f9bada2375538e5a101f/src/Common/Interfaces/ArticleClientInterface.php#L37) method, your IDE should tell you the required parameter type.

Please let us know if you have any other problems.

Thank you.

**Comment by michalsalon-mall** [comment]
Implemented in [v3.14.0](https://github.com/mallgroup/mpapi-client-php/releases/tag/v3.14.0).
